### PR TITLE
Updating karma to v5+. Should resolve  sabberworm /karma-iframes#14

### PR DESCRIPTION
**Pull request by taulinger**
Note: No support for karma < 5.
Resolves [Updating Karma to v5+ : PromiseContainer is not a constructor](https://github.com/sabberworm/karma-iframes/issues/14)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,14 +6,7 @@ node_js:
...
   - "6"
 env:
-  - KARMA_VERSION=1.3
-  - KARMA_VERSION=1.4
-  - KARMA_VERSION=1.5
-  - KARMA_VERSION=1.6
-  - KARMA_VERSION=1.7
-  - KARMA_VERSION=2.0
-  - KARMA_VERSION=3.0
-  - KARMA_VERSION=3.1
```
Comment (sabberworm) [inline .travis.yml:16]: Would it be too difficult to do a hybrid solution that remains backwards-compatible?

Comment (taulinger) [reply .travis.yml:16]: No, that would be possible, i've already thought about that point. I will add it asap.

Btw: The travis env property for the current version of karma must be "KARMA_VERSION=5.2", right ?
If so, I would also add "KARMA_VERSION=4.4".

File: lib/rewrite-middleware.js [hunk 2]
```diff
@@ -10,18 +11,28 @@ let {nonIncludedFiles, REVERSE_CONTEXT} = require('./framework.js');
...
+
+	} else {
+		let files = fileList.files
+		emitter.on('file_list_modified', (filesParam) => { files = filesParam })
+
+		return {
+			then(...args) {
+				return Promise.resolve(files).then(...args)
+			}
+		}
```
Comment (sabberworm) [inline lib/rewrite-middleware.js:33]: Wouldn’t the following also work:

```suggestion
		return new Promise(resolve => {
			emitter.on('file_list_modified', resolve);
		});
```

?

Comment (taulinger) [reply lib/rewrite-middleware.js:33]: I don't think so. As before, the promise functionality was replicated karma webServer.

Comment (sabberworm) [reply lib/rewrite-middleware.js:33]: Ah, yes, I see

**Comment by taulinger** [comment]
What else do you need to merge this PR?

**Comment by sabberworm** [comment]
@taulinger Nothing, I simply thought I’d wait until the build gets auto-migrated from travis-ci.org to travis-ci.com before I attempt again to fix it…

**Pull request merged**